modules/backend/formwidgets/RecordFinder.php

```diff
@@ -273,7 +273,14 @@ public function getNameValue()
             return null;
         }
 
-        return $this->relationModel->{$this->nameFrom};
+        $name = $this->relationModel;
+        $nameFrom = explode('.', $this->nameFrom);
+        foreach ($nameFrom as $attribute) {
+            if (!empty($name->{$attribute})) {
+                $name = $name->{$attribute};
+            }
+        }
```

Comment:
FYI there's definitely a bug here in that if I say `nameFrom: property.location.name`, and `property.location.name` is empty but `property.name` isn't then it will select the wrong value.

Reply:
Wouldn't a simple `break` statement in case of an empty value be enough here?

Something along the lines of:

```
foreach ($nameFrom as $attribute) {
    if (!empty($name->{$attribute})) {
        $name = $name->{$attribute};
    } else {
        $name = null;
        break;
    }
}
```

Reply:
@msimkunas probably yes, if @alpapan doesn't want to continue this PR then perhaps you can resubmit it.